src/Gelf/Transport/AmqpTransport.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the php-gelf package.
+ *
+ * (c) Benjamin Zikarsky <http://benjamin-zikarsky.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Gelf\Transport;
+
+use AMQPExchange;
+use AMQPQueue;
+use Gelf\Encoder\JsonEncoder as DefaultEncoder;
+use Gelf\MessageInterface as Message;
+
+/**
+ * Class AmqpTransport
+ *
+ * @package Gelf\Transport
+ * @see http://php.net/manual/pl/book.amqp.php
+ */
+class AmqpTransport extends AbstractTransport
+{
+    /**
+     * @var AMQPExchange $exchange
+     */
+    protected $exchange;
+
+    /**
+     * @var AMQPQueue $exchange
+     */
+    protected $queue;
+
+    /**
+     * @param AMQPExchange $exchange
+     * @param AMQPQueue $queue
+     */
+    public function __construct(AMQPExchange $exchange, AMQPQueue $queue)
+    {
+        $this->queue = $queue;
+        $this->exchange = $exchange;
+        $this->messageEncoder = new DefaultEncoder();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function send(Message $message)
+    {
+        $rawMessage = $this->getMessageEncoder()->encode($message);
+        $this->exchange->publish(
+            $rawMessage,
+            $this->queue->getName(),
+            0,
+            array(
+                'delivery_mode' => 2,
```

Comment:
This only makes sense with `$queue->setFlags(AMQP_DURABLE);`, doesn't it? 
Would it be better to only add `delivery_mode = 2` if `($this->queue->getFlags() & AMQP_DURABLE) > 0`?